### PR TITLE
Limit addresses to points

### PR DESCRIPTION
**Pull request by rmarianski**
Connects to https://github.com/mapzen/vector-datasource/issues/834

@zerebubuth could you review please?

Migration in prod will be to create sql functions again, and run:

```sql
UPDATE
  planet_osm_polygon
  SET
    mz_building_min_zoom = mz_calculate_min_zoom_buildings(planet_osm_polygon.*)
  WHERE
    "addr:housenumber" IS NOT NULL
     AND COALESCE(mz_building_min_zoom, 999) <> COALESCE(mz_calculate_min_zoom_buildings(planet_osm_polygon.*), 999);
```

### CODE REVIEW

File: yaml/buildings.yaml [hunk 1]
```diff
@@ -106,6 +106,9 @@ filters:
   # address points
   - filter:
       "addr:housenumber": true
+      way:
```
Comment (nvkelso) [inline yaml/buildings.yaml]: This looks to limit it to address nodes only?

What about adding 2nd filter like:

```
- "addr:housenumber": true
- building: true
```

Comment (zerebubuth) [inline yaml/buildings.yaml]: We should have already picked up anything with a `building` tag in an earlier filter. And the `building_standard_properties` includes the house number and street, so it should be only the points that we need to filter in at this stage.

**Comment by zerebubuth** [comment]
Looks good to me! And tests, too :smile: :+1:

File: yaml/buildings.yaml (hunk 1 above)
Comment (rmarianski) [inline yaml/buildings.yaml]: Do we want to drop addresses unless they are on a building? According to [taginfo](http://taginfo.openstreetmap.org/keys/addr:housenumber#combinations), ~ 60% don't have a `building` tag.

Comment (nvkelso) [inline yaml/buildings.yaml]: No, we don't want to drop address points if they are not on a building.

Comment (nvkelso) [inline yaml/buildings.yaml]: I see the post processing transform now, thanks. LGTM.

**Pull request merged**